### PR TITLE
Unhook Settings Browser `dialog:` method from Morphic

### PR DESCRIPTION
**Pull request by hernanmd**
## TL;DR

This PR introduces a mechanism to unhook the `dialog:` method from Morphic, used in Morphic Settings Browser.

## Details

- The `dialog` callback in `PragmaSetting` no longer requires answering a Morph.
- Introduce `ActionSettingDeclaration` for declarations applied through buttons.
- Introduce the `button:` method in `SettingTreeBuilder,` so we explicitly configure a button for a setting. It is paired with the new `ActionSettingDeclaration`. This means that settings using `#group:` now can use `#button:` to detach the Settings Browser from Morphic, as detailed in #16801.
- The `inputWidget` method used `dialog` bound to Morphic. 
- To keep the Morphic Settings Browser setting functional, `ActionSettingDeclaration` includes `inputWidget,` which should be removed when Morphic is removed. The same applies to the method `realValue`.

### CODE REVIEW

**Comment by Ducasse** [comment]
I read it but I was unclear how it uncouple from morphic (reading linear code is not super efficient). 
Let us the feedback of @jecisc

**Comment by hernanmd** [comment]
> I read it but I was unclear how it uncouple from morphic (reading linear code is not super efficient). Let us the feedback of @jecisc

Hi Stef,
Please have a look at the following image to understand the main change in the PR

xnview_aWlWiA

**Comment by hernanmd** [comment]
Thanks!
